framework/src/modules/chain/transactions/transactions_handlers.js

```diff
@@ -151,8 +151,18 @@ const applyTransactions = (storage, exceptions) => async (transactions, tx) => {
 		tx,
 	});
 
-	await Promise.all(transactions.map(t => t.prepare(stateStore)));
-	await Promise.all(transactions.map(t => votesWeight.prepare(stateStore, t)));
+	// Avoid merging both prepare statements into one for...of loop as this slows down the call dramatically
+	// eslint-disable-next-line no-restricted-syntax
+	for (const transaction of transactions) {
+		// eslint-disable-next-line no-await-in-loop
+		await transaction.prepare(stateStore);
+	}
+
+	// eslint-disable-next-line no-restricted-syntax
+	for (const transaction of transactions) {
+		// eslint-disable-next-line no-await-in-loop
+		await votesWeight.prepare(stateStore, transaction);
```

Comment:
I think this also happens in the `undo` case too.
Not important for this PR, but it would be faster to change to `await votesWeight.prepare(stateStore, transactions);`

Reply:
Done